lib/administrate/namespace.rb

```diff
@@ -25,5 +25,30 @@ def all_controller_paths
         route.defaults[:controller].to_s
       end
     end
+
+    class Resource
+      attr_reader :namespace, :resource
+
+      def initialize(namespace, resource)
+        @namespace = namespace
+        @resource = resource
+      end
+
+      def to_s
+        name.to_s
+      end
+
+      def to_sym
+        name
+      end
+
+      def name
+        resource.to_s.gsub(/^#{namespace}\//, "").to_sym
+      end
+
+      def path
+        name.to_s.gsub("/", "_")
+      end
+    end
```

Comment:
What do you think of moving this to its own file? This would allow us to write unit tests for it and keep files leaner.